.github/workflows/tag-dist.yaml

```diff
@@ -0,0 +1,34 @@
+name: Tag Dist Build
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - "v*"
+
+jobs:
+  tag-dist:
+    runs-on: ubuntu-latest
+    needs: ["ci"]
```

Comment:
This might fail. besides it, looks fine, feel free to merge it once this minor thing is fixed

Reply:
good catch.  fixed.